framework/src/utils/BilinearInterpolation.C

```diff
@@ -56,18 +56,30 @@ BilinearInterpolation::getNeighborIndices(const std::vector<Real> & inArr,
   }
 }
 
+Real
+BilinearInterpolation::sample(Real s1, Real s2) const
+{
+  return sampleInternal(s1, s2);
+}
+
+ADReal
+BilinearInterpolation::sample(const ADReal & s1, const ADReal & s2) const
+{
+  return sampleInternal(s1, s2);
+}
+
 template <typename T>
 T
-BilinearInterpolation::sample(const T & xcoord, const T & ycoord) const
+BilinearInterpolation::sampleInternal(T & s1, T & s2) const
```

Comment:
```suggestion
BilinearInterpolation::sampleInternal(const T & s1, const T & s2) const
```
we're not modifying these are we? Would be a little concerning if we are

Reply:
yes that would be an issue